jpype/_core.py

```diff
@@ -95,48 +95,84 @@ def _hasClassPath(args):
 
 _JVM_started = False
 
-def startJVM(jvm=None, *args, **kwargs):
+def startJVM(*args, **kwargs):
     """
     Starts a Java Virtual Machine.  Without options it will start
     the JVM with the default classpath and jvm.  The default classpath
     will be determined by ``jpype.getClassPath()``.  The default JVM is 
     determined by ``jpype.getDefaultJVMPath()``.
 
-    Args:
-      jvm (Optional, str):  Path to the jvm library file,
-        The default of None will use ``jpype.getDefaultJVMPath()``
+    Parameters:
+     *args (Optional, str[]): Arguments to give to the JVM.
+        The first argument may be the path the JVM.
+
+    Keyword Arguments:
+      jvm (str):  Path to the jvm library file,
         Typically one of (``libjvm.so``, ``jvm.dll``, ...) 
-      *args (Optional, str[]): Arguments to give to the JVM
-      classpath (Optional[string]): Set the classpath for the jvm.
+        Default of None will use ``jpype.getDefaultJVMPath()``
+      classpath (str,[str]): Set the classpath for the jvm.
         This will override any classpath supplied in the arguments
-        list.  Default will use ``jpype.getClassPath``
-      ignoreUnrecognized (Optional, [bool]): Option to JVM to ignore
-        invalid JVM arguments.  Default is False.
-    """
+        list. Default will use ``jpype.getClassPath``
+      ignoreUnrecognized (bool): Option to JVM to ignore
+        invalid JVM arguments. Default is False.
+
+    Raises:
+      OSError: if the JVM cannot be started or is already running.
+      TypeError: if an invalid keyword argument is supplied 
+        or a keyword argument conflicts with the arguments.
+
+     """
     if _jpype.isStarted():
          raise OSError('JVM is already started')
     global _JVM_started
     if _JVM_started:
          raise OSError('JVM cannot be restarted')
     _JVM_started = True
 
-    if jvm is None:
-        jvm = getDefaultJVMPath()
-
-        # Check to see that the user has not set the classpath
-        # Otherwise use the default if not specified
-        if not _hasClassPath(args) and 'classpath' not in kwargs:
-            kwargs['classpath'] = _classpath.getClassPath()
+    args = list(args)
 
-    if 'ignoreUnrecognized' not in kwargs:
-        kwargs['ignoreUnrecognized'] = False
+    # JVM path
```

Comment:
this voodoo is not necessary with the old signature. If jvm is already passed via the named keyword, passing it again via kwargs will raise a TypeError.